src/main/java/org/opensearch/security/securityconf/impl/v7/ConfigV7.java

```diff
@@ -497,10 +521,125 @@ public void setEncryptionKey(String encryptionKey) {
             this.encryptionKey = encryptionKey;
         }
 
+        public String getAlgorithm() {
```

Comment:
The changes here are identical tot the ones in ConfigV6, you can make something like a utility class:

```
 public class KeyProperties {
    @JsonProperty
    private String algorithm;
    
    // Common properties for all types
    @JsonProperty("signing_key")
    private String signingKey;
    
    // Additional properties for EC type
    @JsonProperty("ec_private_key")
    private String ecPrivateKey;
    @JsonProperty("ec_private")
    private String ecPrivate;
    @JsonProperty("ec_x")
    private String ecXCoordinate;
    @JsonProperty("ec_y")
    private String ecYCoordinate;
    
    // Additional properties for RSA type
    @JsonProperty("rsa_private_key")
    private String rsaPrivateKey;
    @JsonProperty("rsa_modulus")
    private String rsaModulus;
    @JsonProperty("rsa_public_exp")
    private String rsaPublicExp;
    @JsonProperty("rsa_private_exp")
    private String rsaPrivateExp;
    @JsonProperty("rsa_first_prime_factor")
    private String rsaFirstPrimeFactor;
    @JsonProperty("rsa_second_prime_factor")
    private String rsaSecondPrimeFactor;
    @JsonProperty("rsa_first_prime_crt")
    private String rsaFirstPrimeCrt;
    @JsonProperty("rsa_second_prime_crt")
    private String rsaSecondPrimeCrt;
    @JsonProperty("rsa_first_crt_coefficient")
    private String rsaFirstCrtCoefficient;
    
    // Getters and setters
    // ...
    
    public String getSigningKey() {
        return signingKey;
    }

    // Other common methods
    // ...
}
``` and then have both wrap the utility class. 

Your call however.